src/ferny/session.py

```diff
@@ -139,8 +139,6 @@ async def connect(self,
         if handle_host_key and has_feature('KnownHostsCommand'):
             args.extend([
                 '-o', f'KnownHostsCommand={askpass_path} %I %H %t %K %f',
-                '-o', 'GlobalKnownHostsFile=none',
-                '-o', 'UserKnownHostsFile=none',
```

Comment:
Turns out we don't strictly require this for cockpit with the current API, see https://github.com/cockpit-project/cockpit/pull/18856 . I still find it conceptually cleaner, though.

Reply:
I think the conceptual idea here was that if we have `handle_host_key` then we're taking over *all* control of handling the host key.

Part of my thinking here is that we're overriding the `KnownHostsCommand`.  At some point, that might have an impact on some system whereby we can no longer connect to a host because we don't gets its IPA key (or whatever).  Doing the same for the "files" approach was meant to make it clear that this behaviour is intended.

The other reason I did that was that, in cockpit, by the time we decide to run ferny in this mode, we know that we want the connection to fail with the hostkey in the `close` message.  This guarantees that it fails (since we return False).

All that being said, it counter-acts the documentation on the `do_host_key` callback, which states:

> Returning False means that ssh implements its default logic.

In short, I'm ambivalent about this change.  If you read the above and are still convinced that the change is correct, let's do it.  But in any case: one way or the other, this code is probably not going to survive the "big API change" we've been talking about.

Reply:
> one way or the other, this code is probably not going to survive the "big API change" we've been talking about.

I'm counting on this! :grin: This is tricky to pin down in a consistent way, and I'm really looking forward to dropping both `KnownHostsCommand` and these options, and just let SSH do its thing. For that reason I also don't have a strong opinion about this commit. As you wrote (and my cockpit PR confirms), we don't need it for Cockpit, so it's not on the critical path. I just feel uneasy about it as the unit tests cannot actually cover the effect of these, as they set `UserKnownHostsFile` to a real file, and assert that `handle_known_hosts=True` will actually be able to tell the cases apart. So I'm still leaning towards landing it, as it will make the unit tests more "honest" and useful.

Reply:
Fine by me.